src/Transformation/README.md

```diff
@@ -0,0 +1,32 @@
+ # Transformations
+
+This service abstracts transformations between datatypes and provides some basic
+transformations to be reused throughout the system.
+
+A transformation is a function from one datatype to another. It MUST NOT perform
+any sideeffects, i.e. it must be morally impossible to observe how often the
+transformation was actually performed. It MUST NOT touch the provided value, i.e.
+it is allowed to create new values but not to modify existing values. This would
+be an observable sideeffect.
+
+The actual usage of this interface is quite boring, but we could typehint on
+`Transformation` to announce we indeed want some function having the aforementioned
+properties.
+
+```php
+
+require_once(__DIR__."\Factory.php");
+
+$f = new \ILIAS\Transformation\Factory;
+
+$int_to_string = $f->custom(function ($v) {
+	if (!is_int($v)) {
+		throw new \InvalidArgumentException("Expected int, got ".get_type($v));
+	}
+	return "$v";
+});
```

Comment:
Is it a goal to provide as many of those functions through the factory? E.g. if one needs a new "custom" transformation that is suspected to be useable in other scenario, would we require to add this transformation to the factory? If so, maybe provide a SHOULD guideline encouraging to do so.